src/base/system_parameter.c

```diff
@@ -2849,7 +2849,7 @@ static SYSPRM_PARAM prm_Def[] = {
    (DUP_PRM_FUNC) NULL},
   {PRM_ID_LOG_ISOLATION_LEVEL,
    PRM_NAME_LOG_ISOLATION_LEVEL,
-   (PRM_FOR_CLIENT | PRM_USER_CHANGE | PRM_FOR_SESSION),
+   (PRM_FOR_CLIENT | PRM_FOR_SERVER | PRM_USER_CHANGE | PRM_FOR_SESSION),
```

Comment:
As I know PRM_FOR_SERVER means that cub_server follows the value in the cubrid.conf. If dba set the isolation_level=6(serializable) in the cubrid.conf and start server, then cub_server is configured the isolation_level to the 6 instead of 4(default, read committed). Currently this property is only used on the client side when I check the code. And the manual also said it is a client parameter. Please let me know the reason.

Reply:
@hornetmj 

When changing the value with `set system parameter`, determined with macro `PRM_IS_FOR_SESSION` whether to change the value of `prm_Def` or `session parameter`.
At this time, the true condition of macro `PRM_IS_FOR_SESSION` is `PRM_FOR_SESSION` flag.

And `prm_get_value ()` called by `prm_get_integer_value ()` determined with macro `PRM_SERVER_SESSION` whether the value is read from `prm_Def` or `session parameter`.
At this time, the true condition of macro `PRM_SERVER_SESSION` is `PRM_FOR_SESSION & PRM_IS_FOR_SERVER & !PRM_CLIENT_SESSION_ONLY` flag.

Therefore, `PRM_ID_LOG_ISOLATION_LEVEL` and `PRM_ID_LK_TIMEOUT` that do not have `PRM_IS_FOR_SERVER` flag are taken from `prm_Def`, not sesson.

I added `PRM_FOR_SERVER` flag to solve this.



But, to solve this problem without adding the flag, i think that need to make an exception in the conditional statement below.

```
if ((PRM_SERVER_SESSION (prm_id) || (something...)) && BO_IS_SERVER_RESTARTED())
{
        SESSION_PARAM *sprm;
        thread_p = thread_get_thread_entry_info();
        sprm = session_get_session_parameter(thread_p, prm_id);
        If (sp.m)
        {
                return &(sprm->value);
        }

       .....
}
```

Reply:
@hornetmj 

In the manual book `시작하세요! 큐브리드`, I found that `lock_timeout` and `isolation_level` can be set by `cubrid.conf`. Isn't it valid method now? (the first edition of this book is dated 2015.)

Reply:
@Yechan0815 

> @hornetmj
> 
> In the manual book `시작하세요! 큐브리드`, I found that `lock_timeout` and `isolation_level` can be set by `cubrid.conf`. Isn't it valid method now? (the first edition of this book is dated 2015.)

It is true. The client or server or client/server system paramters can be set in the cubrid.conf.

Reply:
From my point of view, the following two currently feel strange:

1. For the same session parameter(e.g. lock_timeout), why should setter and getter decide access to prm_Def or session parameter through different conditions?
    - Wouldn't it be the right approach to modify the setter and getter to use the same condition when accessing the session parameter?
2. The lock_timeout and isolation_level have no effect on server (overall) operation. It only has a limited effect on a particular session. The manual describes the server parameter as follows.
    - A server parameter affects the behaviors of the server processes.

I understand that your solution solves the reported problem. However, for the above reasons, I would like to hear more from you and the reviewers.

Reply:
@hornetmj 

1.
It feels strange to me that the conditions of getter and setter are different. But I can't explain this part because I haven't figured out the designer's intentions yet...

2.
Both the `tdes` initial value and the `sessoin` initial value use the value sent by the client at boot. The client sends `-1 (infinite)` and `4 (read commited)` as shown below.
```c
tran_isolation = TRAN_DEFAULT_ISOLATION_LEVEL ();
tran_lock_wait_msecs = TRAN_LOCK_INFINITE_WAIT;
```
In fact, even if `prm_Def` is set by `cubrid.conf`, it does not effect on  `tdes` or `session`.

Reply:
@hornetmj 

> The lock_timeout and isolation_level have no effect on server (overall) operation. It only has a limited effect on a particular session. The manual describes the server parameter as follows.
A server parameter affects the behaviors of the server processes.

By the definition of `PRM_SERVER_SESSION()` macro and its usage in `prm_get_value()`, I guess that these kinds of parameters **have different values with each session** and are **used in the server process**. I translate "A server parameter affects the behaviors of the server processes." as "The parameter is used by server processes". In this respect, I think this makes enough sense.

Not `PRM_SERVER_SESSION()`, but `PRM_FOR_SESSION` examples:
`optimization_level`, `csql_history_num`, `csql_auto_commit`, and `multi_range_optimization_limit`.

These are only used on the client process like CAS, Csql, or JSP. On the other hand `PRM_SERVER_SESSION()` parameters are used also on the server process. The example is `create_table_reuseoid`, `timezone`, and `alter_table_change_type_strict`.

However, there are some exceptions to my guess such as `no_backslash_escapes`, which seems to be used only on the client process but has `PRM_FOR_SERVER`. I think this flag setting is wrong, though.

This is just my opinion based on the code and the usage. What do you think?

Reply:
> For the same session parameter(e.g. lock_timeout), why should setter and getter decide access to prm_Def or session parameter through different conditions?

Regarding this, I don't know exactly what you mean by the setter's condition?

Reply:
@Yechan0815 
> In fact, even if prm_Def is set by cubrid.conf, it does not effect on tdes or session.

If the `lock_timeout` is set by cubrid.conf, they are applied when a session is connected in the `boot_restart_client()` and work well. Aren't they? Or did I misunderstand what you mean?

Reply:
@Yechan0815 On top of that, I have a question. If we set PRM_FOR_SERVER, so make it use the session variable, the `lock_timeout` on cubrid.conf doesn't work? It's about all server session variables. They can't set by cubrid.conf?

Reply:
@wrlawodms 

The book states that cubrid.conf affects the 'server'. But it works fine for csql.

Reply:
> @Yechan0815 On top of that, I have a question. If we set PRM_FOR_SERVER, so make it use the session variable, the `lock_timeout` on cubrid.conf doesn't work? It's about all server session variables. They can't set by cubrid.conf?

It works fine. I'm sorry.

Reply:
@Yechan0815 no need to sorry. I just wonder the server session parameter can't be set by cubrid.conf. Have you tried and confirmed it works?

Reply:
@wrlawodms 

> > For the same session parameter(e.g. lock_timeout), why should setter and getter decide access to prm_Def or session parameter through different conditions?
> 
> Regarding this, I don't know exactly what you mean by the setter's condition. Could you elaborate a little more?

It seems that getter is saying `prm_get_value ()` and setter is `sysprm_set_value ()`.

The condition that the getter `prm_get_value ()` accesses the session is `PRM_SERVER_SESSION ()`, and the condition that the setter `sysprm_set_value ()` accesses the session is `PRM_IS_FOR_SESSION ()`.
So the conditions for getter and setter are different.

Reply:
@wrlawodms 

> @Yechan0815 no need to sorry. I just wonder if the server session parameter can't be set by cubrid.conf. Have you tried and confirmed it works?

Yes it works fine.

Reply:
> @hornetmj
> 
> > The lock_timeout and isolation_level have no effect on server (overall) operation. It only has a limited effect on a particular session. The manual describes the server parameter as follows.
> > A server parameter affects the behaviors of the server processes.
> 
> By the definition of `PRM_SERVER_SESSION()` macro and its usage in `prm_get_value()`, I guess that these kinds of parameters **have different values with each session** and are **used in the server process**. I translate "A server parameter affects the behaviors of the server processes." as "The parameter is used by server processes". In this respect, I think this makes enough sense.
> 
> Not `PRM_SERVER_SESSION()`, but `PRM_FOR_SESSION` examples: `optimization_level`, `csql_history_num`, `csql_auto_commit`, and `multi_range_optimization_limit`.
> 
> These are only used on the client process like CAS, Csql, or JSP. On the other hand `PRM_SERVER_SESSION()` parameters are used also on the server process. The examples are `create_table_reuseoid`, `timezone`, and `alter_table_change_type_strict`.
> 
> However, there are some exceptions to my guess such as `no_backslash_escapes`, which seems to be used only on the client process but has `PRM_FOR_SERVER`. I think this flag setting is wrong, though.
> 
> This is just my opinion based on the code and the usage. What do you think?

I agree with you and @joohok(off-line) feedback in aspects of the implemenation of this module. However, I still have doubts about the question posed above. I approve of this resolution given the examples you provided and in terms of addressing the problem.

Reply:
It also looks weird to me that the condition for setting and getting is different. @Yechan0815 Please hold the issue a little bit more.

Reply:
@Yechan0815 I've taken a look at the system parameter source code to know a reasonable rule of its flags, but there seems to be some inconsistent use and behavior to build a consistent rule, at least to me. To get a precise concept of that, I think we should use more time and effort to analyze it. Regarding this issue, I suppose It would be better just to resolve it here and analyze it when a problem occurs, or it could be the following job after merging this issue. Anyway, I approved merging this PR.

Reply:
@wrlawodms 
I keep reading, but it's difficult. Thank you!